tests/core/_plot.py

```diff
@@ -0,0 +1,213 @@
+# FILE: tests/core/test_plot.py
+
+import pytest
+import os
+from unittest.mock import MagicMock
+from typing import Dict, List, Any
+
+from gesund.core._plot import CommonPlots
+from gesund.core import UserInputParams, UserInputData
+
+
+@pytest.fixture
+def common_plots() -> CommonPlots:
+    return CommonPlots()
+
+
+@pytest.fixture
+def mock_cls_driver(common_plots: CommonPlots) -> MagicMock:
+    mock = MagicMock()
+    common_plots.cls_driver = mock
+    return mock
+
+
+@pytest.fixture
+def mock_obj_driver(common_plots: CommonPlots) -> MagicMock:
+    mock = MagicMock()
+    common_plots.obj_driver = mock
+    return mock
+
+
+@pytest.fixture
+def mock_seg_driver(common_plots: CommonPlots) -> MagicMock:
+    mock = MagicMock()
+    common_plots.seg_driver = mock
+    return mock
+
+
+def test_common_plots_initialization(common_plots: CommonPlots) -> None:
+    """Test that all driver attributes are initialized to None."""
+    assert common_plots.cls_driver is None
+    assert common_plots.obj_driver is None
+    assert common_plots.seg_driver is None
+
+
+def test_class_distribution_plot(common_plots: CommonPlots, mock_cls_driver: MagicMock) -> None:
+    """Test the _class_distribution method calls the cls_driver's _plot_class_distributions correctly."""
+    metrics: Dict[str, int] = {"classA": 100, "classB": 150}
+    threshold: float = 0.5
+    save_path: str = "./plots/class_distribution.png"
```

Comment:
@ozkanuysal the save path should be a temporary directory, please check https://docs.python.org/3/library/tempfile.html, so you can use the tempdir + with context manager, so that the tempdir is deleted after the context execution is completed.